pkg/utils/db.go

```diff
@@ -52,6 +52,89 @@ func GetFlavor(ctx context.Context, db *sql.DB) (string, error) {
 	return gmysql.MySQLFlavor, nil
 }
 
+// GetAllServerID gets all slave server id and master server id
+func GetAllServerID(ctx context.Context, db *sql.DB) (map[uint32]struct{}, error) {
+	serverIDs, err := GetSlaveServerID(ctx, db)
+	if err != nil {
+		return nil, err
+	}
+
+	masterServerID, err := GetServerID(db)
+	if err != nil {
+		return nil, err
+	}
+
+	serverIDs[masterServerID] = struct{}{}
+	return serverIDs, nil
+}
+
+// GetSlaveServerID gets all slave server id
+func GetSlaveServerID(ctx context.Context, db *sql.DB) (map[uint32]struct{}, error) {
+	rows, err := db.QueryContext(ctx, `SHOW SLAVE HOSTS`)
+	if err != nil {
+		return nil, terror.DBErrorAdapt(err, terror.ErrDBDriverError)
+	}
+	defer rows.Close()
+
+	rowColumns, err := rows.Columns()
+	if err != nil {
+		return nil, terror.DBErrorAdapt(err, terror.ErrDBDriverError)
+	}
+
+	/*
+		in MySQL:
+		mysql> SHOW SLAVE HOSTS;
+		+------------+-----------+------+-----------+--------------------------------------+
+		| Server_id  | Host      | Port | Master_id | Slave_UUID                           |
+		+------------+-----------+------+-----------+--------------------------------------+
+		|  192168010 | iconnect2 | 3306 | 192168011 | 14cb6624-7f93-11e0-b2c0-c80aa9429562 |
+		| 1921680101 | athena    | 3306 | 192168011 | 07af4990-f41f-11df-a566-7ac56fdaf645 |
+		+------------+-----------+------+-----------+--------------------------------------+
+
+		in MariaDB:
+		mysql> SHOW SLAVE HOSTS;
+		+------------+-----------+------+-----------+
+		| Server_id  | Host      | Port | Master_id |
+		+------------+-----------+------+-----------+
+		|  192168010 | iconnect2 | 3306 | 192168011 |
+		| 1921680101 | athena    | 3306 | 192168011 |
+		+------------+-----------+------+-----------+
+	*/
+
+	var (
+		serverID  sql.NullInt64
+		host      sql.NullString
+		port      sql.NullInt64
+		masterID  sql.NullInt64
+		slaveUUID sql.NullString
+	)
+	serverIDs := make(map[uint32]struct{})
+	for rows.Next() {
+		if len(rowColumns) == 5 {
+			err = rows.Scan(&serverID, &host, &port, &masterID, &slaveUUID)
+		} else {
+			err = rows.Scan(&serverID, &host, &port, &masterID)
+		}
+		if err != nil {
+			return nil, terror.DBErrorAdapt(err, terror.ErrDBDriverError)
+		}
+
+		if serverID.Valid {
+			serverIDs[uint32(serverID.Int64)] = struct{}{}
```

Comment:
If `serverID.Int64` exceeds math.MaxUint32, it will overflow and be transferred to an unexpected number. 
Run following codes:
```go
	var a int64 = math.MaxUint32 + 1
	var b = uint32(a)
	fmt.Println(b)
```
will get 0.

Reply:
the server id is select from mysql, in fact it will not execeed the math.MaxUint32